### PR TITLE
Music Genre Classification

### LINKED ISSUE #576: A trained model for Music Genre Classification
**Is your feature request related to a problem? Please describe.**
A pre-trained model for music genre classification is not tough to find. A good model for music genre classification is like a needle in haystack.

**Describe the solution you'd like**
After working on Music classification for more than a year, I believe I can come up with a really good model. I will be using  Benchmark dataset, `GTZAN`, `Emotify` for example, Python offers TF, Keras and most importantly  `pyaudioanalysis` so i think Python is the best choice.

**Describe alternatives you've considered**
There are multiple models available, but either they are now old, they are using old algos like `SVM`, or old variants of Neighbours. 

**Additional context**

### PR DESCRIPTION
**Pull request by vybhav72954**
# Description

Deep Learning Implementation of Music Genre Classification.

Fixes #576 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

File: Python/Music_Genre_Classification/feature_extraction.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# The Genres in Dataset
+genres = 'blues classical country disco hiphop jazz metal pop reggae rock'.split()
+
+for g in genres:
+    # Traversing through various genres in the Dataset
+    # Feed the complete path of the GTZAN folder
+    for filename in os.listdir(f'Database/GTZAN/{g}'):
+
+        # Traversing through various songs in a particular genre.
+        songname = f'C:/Users/asus/PycharmProjects/Music_Classification/Database/GTZAN/{g}/{filename}'
```
Comment (HarshCasper) [inline Python/Music_Genre_Classification/feature_extraction.py:35]: Can you remove all local references? It is preferred to use absolute paths

File: Python/Music_Genre_Classification/README.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+275/275 [==============================] - 5s 20ms/step - loss: 0.6635 - acc: 0.7932 - val_loss: 0.9442 - val_acc: 0.7273
+```
+Here the attained accuracy is closer to 80%.
+
+## Author(s)  
+
+Made by [Vybhav Chaturvedi](https://www.linkedin.com/in/vybhav-chaturvedi-0ba82614a/)
+
+If you like this script, I have made Repos related to Music Classification, Music Mood Classification, Music Human Emotion mapping,
+Music Feature Classification.
```
Comment (HarshCasper) [inline Python/Music_Genre_Classification/README.md:150]: Reference the repos as well. Would be helpful to people to quickly find them.

File: Python/Music_Genre_Classification/README.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+- pandas
+- scikit-learn
+- scipy
+
+Considering there are some big libraries here, it is recommended to setup a virtual environment first. This
+[link](https://realpython.com/lessons/creating-virtual-environment/) might help.
+
+## Dataset
+
+We have used GTZAN dataset for our task. It is publicly available, here is the
```
Comment (HarshCasper) [inline Python/Music_Genre_Classification/README.md:34]: Since you have already referenced the GTZAN Dataset, do we actually need to store it in the repo as well?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few Changes: 

- Remove the Database 
- Upload the Model File on G-Drive/Dropbox and drop the link here
- Make changes on Python README as well

File: Python/Music_Genre_Classification/feature_extraction.py (hunk 1 above)
Comment (vybhav72954) [reply Python/Music_Genre_Classification/feature_extraction.py:35]: What's wrong with me, I have been pushing a lot of local addresses lately, it skips my mind for some reason 😞

Comment (HarshCasper) [reply Python/Music_Genre_Classification/feature_extraction.py:35]: No worries! You are doing a great work.

File: Python/Music_Genre_Classification/README.md (hunk 3 above)
Comment (vybhav72954) [reply Python/Music_Genre_Classification/README.md:34]: I don't get that, I have just referenced the Repo here, and have stored a file containing 6 features, the whole DB is 1.2 gb.

Can you clarify this?

Comment (vybhav72954) [reply Python/Music_Genre_Classification/README.md:34]: The size of the `csv` is also half MB, if you want I can get rid of the file and paste a drive link there

**Comment by vybhav72954** [comment]
@HarshCasper Would it be fine if I update `README` in a later PR (~not a separate PR~ but as in when I will solve other issues assigned to me here), I made the branch in my local before the changes in README were made so I still have the older version, also my local is kind of messed up I have multiple upstreams and have kind of lost track (cause i had to test-run my actions somewhere), hence I was planning to remove that local after all my open PRs are merged and clone a fresh one. Would be great if you can allow that.

**Comment by vybhav72954** [comment]
> * Make changes on Python README as well

Fixed in #720

File: Python/Music_Genre_Classification/README.md (hunk 3 above)
Comment (HarshCasper) [reply Python/Music_Genre_Classification/README.md:34]: Yeah. The only intent was to get rid of the files that can be retrieved from online sources.

**Comment by HarshCasper** [comment]
Thanks for the PR! 

I am inclined towards merging the same, but can you take a look at the failing checks and the reasons they are failing. As far as I can notice, the Spell Check Action is generating a lot of unnecessary noise. The LinkedIn URL also directs to 999 Status Code, which is understandable since LinkedIn filter requests based on the user-agent. Can you also try looking at the other errors and possibly fix them or raise an Issue to be fixed in these actions?

**Comment by vybhav72954** [comment]
Can you check #672 @HarshCasper

**Review by HarshCasper — APPROVED** [review]
